src/engine/NodeHelpers.cs

```diff
@@ -97,13 +97,16 @@ public static void FreeChildren(this Node node, bool detach = false)
     /// </remarks>
     public static void ReParent(this Node node, Node newParent)
     {
-        if (node.GetParent() == null)
+        lock (node.GetParent())
```

Comment:
Why is there a lock here? Godot operations *must* always happen from the main thread, so this is completely useless, or at least should be. If background stuff needs to touch things here it should go through Invoke.